packages/1155-contracts/package/premint-api.ts

```diff
@@ -0,0 +1,330 @@
+    zoraBackendChainName: BackendChainNames.ZORA_TESTNET,
+  },
+  [foundry.id]: {
+    chainId: foundry.id,
+    isTestnet: true,
+    zoraPathChainName: "zora",
+    zoraBackendChainName: BackendChainNames.ZORA_TESTNET,
+  }
+};
+
+type MintArgumentsSettings = {
+  tokenURI: string;
+  maxSupply?: bigint;
+  maxTokensPerAddress?: bigint;
+  pricePerToken?: bigint;
+  mintStart?: bigint;
+  mintDuration?: bigint;
+  royaltyMintSchedule?: number;
+  royaltyBPS?: number;
+  royaltyRecipient?: Address;
+  fixedPriceMinter?: Address;
+};
+
+const OPEN_EDITION_MINT_SIZE = "18446744073709551615";
+const DefaultMintArguments = {
+  maxSupply: BigInt(OPEN_EDITION_MINT_SIZE),
+  maxTokensPerAddress: 0n,
+  pricePerToken: 0n,
+  mintDuration: BigInt(60 * 60 * 24 * 7), // 1 week
+  mintStart: 0n,
+  royaltyMintSchedule: 0,
+  royaltyBPS: 1000, // 10%,
+};
+
+type PremintResponse = {
+  collection: {
+    contractAdmin: Address;
+    contractURI: string;
+    contractName: string;
+  };
+  premint: {
+    tokenConfig: {
+      tokenURI: string;
+      maxSupply: string;
+      maxTokensPerAddress: string;
+      pricePerToken: string;
+      mintStart: string;
+      mintDuration: string;
+      royaltyMintSchedule: number;
+      royaltyBPS: number;
+      royaltyRecipient: Address;
+      fixedPriceMinter: Address;
+    };
+    uid: number;
+    version: number;
+    deleted: boolean;
+  };
+  chain_name: BackendChainNames;
+  signature: Hex;
+};
+
+const convertPremint = (premint: PremintResponse["premint"]) => ({
+  ...premint,
+  tokenConfig: {
+    ...premint.tokenConfig,
+    maxSupply: BigInt(premint.tokenConfig.maxSupply),
+    pricePerToken: BigInt(premint.tokenConfig.pricePerToken),
+    mintStart: BigInt(premint.tokenConfig.mintStart),
+    mintDuration: BigInt(premint.tokenConfig.mintDuration),
+    maxTokensPerAddress: BigInt(premint.tokenConfig.maxTokensPerAddress),
+  },
+});
+
+const encodePremintForAPI = ({ tokenConfig, ...premint }: PremintConfig) => ({
+  ...premint,
+  tokenConfig: {
+    ...tokenConfig,
+    maxSupply: tokenConfig.maxSupply.toString(),
+    pricePerToken: tokenConfig.maxSupply.toString(),
+    mintStart: tokenConfig.mintStart.toString(),
+    mintDuration: tokenConfig.mintDuration.toString(),
+    maxTokensPerAddress: tokenConfig.maxTokensPerAddress.toString(),
+  },
+});
+
+const ZORA_PREMINT_API_BASE = "https://api.zora.co/premint/";
+
+export class PreminterAPI {
+  network: NetworkConfig;
+  chain: Chain;
+  mintFee: BigInt;
+  constructor(chain: Chain) {
+    this.chain = chain;
+    const networkConfig = networkConfigByChain[chain.id];
+    if (!networkConfig) {
+      throw new Error(`Not configured for chain ${chain.id}`);
+    }
+    this.network = networkConfig;
+    this.mintFee = BigInt("777000000000000");
+  }
+
+  async get(path: string) {
+    const response = await fetch(path, { method: "GET" });
+    return await response.json();
+  }
+
+  async post(path: string, data: any) {
```

Comment:
minor, but this should be extracted to a static function outside the class since it doesn't have any dependency on `this`, same as above with `get`.  would clean up the api definition too

Reply:
I want to allow this to be easily overridable

Reply:
what's the reason for someone wanting to override this?